extensions/jira-search/src/issue.ts

```diff
@@ -79,13 +82,37 @@ function buildJql(query: string): string {
   return jql + " order by lastViewed desc"
 }
 
-function jqlFor(query: string): string {
-  return isIssueKey(query) ? `key=${query}` : buildJql(query)
+function jqlForFilter(filter: IssueFilter) {
+  switch (filter) {
+    case "allIssues":
+      return ""
+    case "issuesInOpenSprints":
+      return "sprint in openSprints()"
+    case "myIssues":
+      return "assignee=currentUser()"
+    case "myIssuesInOpenSprints":
+      return "assignee=currentUser() AND sprint in openSprints()"
+    default:
+      throw new Error(`Unknown filter ${filter}`)
```

Comment:
The `default:`-Case isn't necessary here, as the above four cases handle all options allowed by the type `IssueFilter`.

No further comments besides this – PR can be approved.